src/Marten.Testing/performance_tuning.cs

```diff
@@ -18,24 +13,36 @@ public class JilSerializer : ISerializer
         private readonly Options _options
             = new Options(dateFormat: DateTimeFormat.ISO8601, includeInherited:true);
 
-        public void ToJson(object document, TextWriter writer)
+        public void ToJson(object document, Stream stream)
         {
+            using var writer = new StreamWriter(stream);
             JSON.Serialize(document, writer, _options);
+            writer.Flush();
         }
 
         public string ToJson(object document)
         {
             return JSON.Serialize(document, _options);
         }
 
-        public T FromJson<T>(TextReader reader)
+        public T FromJson<T>(Stream stream)
         {
-            return JSON.Deserialize<T>(reader, _options);
+            return JSON.Deserialize<T>(new StreamReader(stream), _options);
         }
 
-        public object FromJson(Type type, TextReader reader)
+        public Task<T> FromJsonAsync<T>(Stream stream, CancellationToken cancellationToken = default)
```

Comment:
We should change the signature of this method to return a ValueTask

Reply:
Used ValueTask in https://github.com/JasperFx/marten/pull/1685/commits/1ef5e80e9875fa88b5cd902e75eca143937d6488.